com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineDesiredStateAchievingActionsStep.java

```diff
@@ -40,11 +39,10 @@ protected StepPhase executeStep(ExecutionWrapper execution) {
         getStepLogger().debug(Messages.CURRENT_STATE, appName, currentState);
         ApplicationStartupState desiredState = computeDesiredState(execution.getContext(), app);
         getStepLogger().debug(Messages.DESIRED_STATE, appName, desiredState);
-        ApplicationStager applicationStager = new ApplicationStager(execution.getControllerClient());
+        boolean isAppStaged = StepsUtil.getUploadToken(execution.getContext()) == null;
```

Comment:
Better naming of the variable.
Also, could you extract `StepsUtil.getUploadToken(execution.getContext())` into separate variable?

Reply:
Done. I've renamed it and inverted it so it is easier to understand.